client-js/app/scripts/app.ts

```diff
@@ -187,7 +183,13 @@ window.addEventListener('WebComponentsReady', () => {
       });
     }
 
+    onNewItem(x) {
+      console.log('got a new item',x);
+      this.globElem.repopulateList();
+    }
+
     stop(): Promise<void> {
+      this.api.unwatch({path: this.path});
```

Comment:
As with watch, this is an api call that returns a promise, so you can just return that result from this method:
```typescript
stop(): Promise<void> {
  return this.api.unwatch({path: this.path});
}
```